src/modules/CSADigitizer/CSADigitizerModule.cpp

```diff
@@ -0,0 +1,337 @@
+    // Enable parallelization of this module if multithreading is enabled
+    enable_parallelization();
+
+    // Require PixelCharge message for single detector
+    messenger_->bindSingle(this, &CSADigitizerModule::pixel_message_, MsgFlags::REQUIRED);
+
+    // Seed the random generator with the global seed
+    random_generator_.seed(getRandomSeed());
+
+    // Set defaults for config variables
+    config_.setDefault<bool>("simple_parametrisation", true);
+    // defaults for the "advanced" parametrisation
+    config_.setDefault<double>("rise_time_constant", Units::get(1e-9, "s"));  
+    config_.setDefault<double>("feedback_time_constant", Units::get(10e-9, "s"));  // R_f * C_f
+    // for both cases
+    config_.setDefault<double>("feedback_capacitance", Units::get(5e-15, "C/V"));
+    // and for the advanced one
+    config_.setDefault<double>("krummenacher_current", Units::get(20e-9, "C/s"));
+    config_.setDefault<double>("detector_capacitance", Units::get(100e-15, "C/V"));
+    config_.setDefault<double>("amp_output_capacitance", Units::get(20e-15, "C/V"));
+    config_.setDefault<double>("transconductance", Units::get(50e-6, "C/s/V") );
+    config_.setDefault<double>("v_temperature", Units::get(25.7e-3, "eV"));  // Boltzmann kT at 298K
+
+    config_.setDefault<double>("amp_time_window", Units::get(0.5e-6, "s"));
+    config_.setDefault<double>("threshold", Units::get(10e-3, "V"));
+    config_.setDefault<double>("sigma_noise", Units::get(0.1e-3, "V"));
+    
+    config_.setDefault<bool>("output_pulsegraphs", false);
+    config_.setDefault<bool>("output_plots", config_.get<bool>("output_pulsegraphs"));
+    config_.setDefault<int>("output_plots_scale", Units::get(30, "ke"));
+    config_.setDefault<int>("output_plots_bins", 100);
+
+
+    // Copy some variables from configuration to avoid lookups:
+    tmax_ = config_.get<double>("amp_time_window");
+    if (config_.get<bool>("simple_parametrisation")){
+      tauF_ = config_.get<double>("feedback_time_constant");
+      tauR_ = config_.get<double>("rise_time_constant");
+      cf_ = config_.get<double>("feedback_capacitance");
+      rf_=tauF_/cf_; 
+      LOG(DEBUG) << "Parameters: cf " << Units::display(cf_/1e-15, "C/V")  << ", rf " << Units::display(rf_, "V*s/C")
+		 << ", tauF_ " << Units::display(tauF_, "s") << ", tauR_ " << Units::display(tauR_, "s");
+    }
+    else{
+      ikrum_ = config_.get<double>("krummenacher_current");
+      cd_ = config_.get<double>("detector_capacitance");
+      cf_ = config_.get<double>("feedback_capacitance");
+      co_ = config_.get<double>("amp_output_capacitance");
+      gm_ = config_.get<double>("transconductance"); 
+      vt_ = config_.get<double>("v_temperature");
+
+      // helper variables: transconductance and resistance in the feedback loop 
+      // weak inversion: gf = I/(n V_t) (e.g. Binkley "Tradeoff and Optimisation in Analog CMOS design")
```

Comment:
Please add relevant references to the manual bibliography and the README, have a look e.g. at GenericPropagation how to do that.